misc-scripts/xref_mapping/XrefParser/HPAParser.pm

```diff
@@ -39,70 +42,55 @@ sub run {
   my $source_id    = $ref_arg->{source_id};
   my $species_id   = $ref_arg->{species_id};
   my $files        = $ref_arg->{files};
-  my $verbose      = $ref_arg->{verbose};
-  my $dbi          = $ref_arg->{dbi};
-  $dbi = $self->dbi unless defined $dbi;
+  my $verbose      = $ref_arg->{verbose} || 0;
+  my $dbi          = $ref_arg->{dbi} || $self->dbi;
 
-  if((!defined $source_id) or (!defined $species_id) or (!defined $files) ){
-    croak "Need to pass source_id, species_id, files and rel_file as pairs";
-  }
-  $verbose |=0;
+  croak "Need to pass source_id, species_id, files and rel_file as pairs"
+    unless defined $source_id and defined $species_id and defined $files;
 
-  my $filename = @{$files}[0];
+  my $file = @{$files}[0];
 
-  my $file_io = $self->get_filehandle($filename);
-  if ( !defined($file_io) ) {
-    return 1;
-  }
+  my $file_io = $self->get_filehandle($file);
+  croak "ERROR: Could not open $file\n" unless defined $file_io;
 
-  my $parsed_count = 0;
+  my $input_file = Text::CSV->new({ sep_char           => ",",
+				    empty_is_undef     => 1
+				  }) or croak "Cannot use file $file: ".Text::CSV->error_diag ();
+  
+  $file_io->getline(); # skip header # Antibody,antibody_id,ensembl_peptide_id,link
 
-  $file_io->getline();
-
-  while ( defined( my $line = $file_io->getline() ) ) {
-
-    $line =~ s/\s*$//;
-		
-
-    my ( $antibody, $antibody_id, $ensembl_peptide_id, $link) = split( /,/, $line );
+  my $parsed_count = 0;
+  while ( my $row = $input_file->getline( $file_io ) ) {
+    map { $_ =~ s/\s*$// } @{$row};
+    my ( $antibody, $antibody_id, $ensembl_peptide_id, $link ) = @{$row};
 
-    if ( !defined($antibody) || !defined($ensembl_peptide_id) ) {
-      printf( "Line %d contains  has less than two columns.\n",
-	      1 + $parsed_count );
-      print ("The parsing failed\n");
-      return 1;
-    }
+    croak sprintf("Line %d contains has less than two columns.\nParsing failed", 1 + $parsed_count)
+      unless defined $antibody and defined $ensembl_peptide_id;
 	
-    my $label       = $antibody;
-    my $type        = 'translation';
-    my $description = '';
-    my $version     = '1';
-
-    ++$parsed_count;
+    my $label   = $antibody;
+    my $type    = 'translation';
+    my $version = '1';
 
     my $xref_id = $self->get_xref( $antibody_id, $source_id, $species_id, $dbi );
+    $xref_id = $self->add_xref({ acc        => $antibody_id,
+				 version    => $version,
+				 label      => $label,
+				 source_id  => $source_id,
+				 species_id => $species_id,
+				 dbi        => $dbi,
+				 info_type  => "DIRECT"} ) unless defined $xref_id or $xref_id ne '';
```

Comment:
Complex postfix condition is important, I would prefer it up front as this is multi-line and multi-term.

Reply:
It's a matter of preference, but I can make you happy :)